btk/survey.py

```diff
@@ -1,145 +1,77 @@
 """Contains information for surveys available in BTK."""
 import os
 import random as rd
-from collections import namedtuple
-from typing import Iterable
+from collections.abc import Callable
 
 import astropy.wcs as WCS
+import galcheat
 import galsim
 import numpy as np
 from astropy.io import fits
-from hydra import compose
-from hydra import initialize
-from omegaconf import OmegaConf
 
 
-Survey = namedtuple(
-    "Survey",
-    [
-        "name",
-        "pixel_scale",  # arcseconds per pixel
-        "effective_area",  # Effective total light collecting area in square meters [m2]
-        "mirror_diameter",  # in meters [m]
-        "airmass",  # Optical path length through atmosphere relative to zenith path length.
-        "filters",
-        "zeropoint_airmass",
-    ],
-)
-
-Survey.__doc__ = """
-Class containing the informations relative to a survey.
-
-Args:
-    name (str): Name of the survey
-    pixel_scale (float): Pixel scale of the survey, in arcseconds per pixel
-    effective_area (float): Effective total light collecting area, in square meters
-    mirror_diameter (float): Diameter of the primary mirror, in meters
-    airmass (float): Optical path length through atmosphere relative to zenith path length
-    filters (list): List of Filter objects corresponding to the filters of this survey
-    zeropoint_airmass (float) Airmass at which the zeropoint is measured"""
-
-Filter = namedtuple(
-    "Filter",
-    [
-        "name",
-        "psf",  # galsim psf model or function to generate it
-        "sky_brightness",  # mags/sq.arcsec
-        "exp_time",  # in seconds [s]
-        "zeropoint",  # in mags
-        "extinction",  # Exponential extinction coefficient for atmospheric absorption.
-    ],
-)
-
-Filter.__doc__ = """
-Class containing the informations relative to a filter (for a specific survey).
-
-Args:
-    name (str): Name of the filter
-    psf: Contains the PSF information, either as a Galsim object,
-          or as a function returning a Galsim object
-    sky_brightness (float): Sky brightness, in mags/sq.arcsec
-    exp_time (int): Total exposition time, in seconds
-    zeropoint (float): Magnitude of an object with a measured flux of 1 electron per second
-    extinction (float): Exponential extinction coefficient for atmospheric absorption"""
-
-
-def get_survey_from_cfg(survey_conf: OmegaConf):
-    """Creates the corresponding `btk.survey.Survey` object using the information from config file.
+def get_surveys(names="LSST", psf_func: Callable = None):
+    """Return specified surveys from galcheat extended to contain PSF information.
```

Comment:
You might want to add a bit more details to the docstring, like the fact that you return a list of modified galcheat.Survey instances and maybe rename the method to something like `get_survey_list`.

Reply:
Plus I guess there should not be a default value for such method.

Reply:
I agree thanks